### PR TITLE
updating CI readme

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes https://github.com/Lightning-AI/lightning/pull/18440#discussion_r1310931440

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/ci-rtfd.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+permissions:
+  pull-requests: write
+
+jobs:
+  documentation-links:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: readthedocs/actions/preview@v1
+        with:
+          project-slug: "torchmetrics"
```
[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci-rtfd.yml:16]: :innocent: thx, my bad

[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci-rtfd.yml:16]:
```suggestion
          project-slug: "pytorch-lightning"
```

File: .github/workflows/README.md [hunk 2]
```diff
@@ -22,41 +32,47 @@
...
 ## Documentation
 
-| workflow file                     | action       |
-| --------------------------------- | ------------ |
-| .github/workflows/docs-checks.yml | Run doctest. |
+| workflow file                                                                               | action                                                                   |
+| ------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------ |
+| .github/workflows/docs-build.yml                                                            | Run doctest, linkcheck and full HTML build.                              |
+| .github/workflows/ci-rtfd.yml                                                               | Append link to the PR describtion with temporaty ReadTheDocs build docs. |
+| .github/workflows/ci-check-md-links.yml <br> .github/workflows%2Fmarkdown.links.config.json | Validate links in markdown files.                                        |
```
Comment (carmocca) [inline .github/workflows/README.md:39]:
```suggestion
| .github/workflows/ci-check-md-links.yml <br> .github/workflows/markdown.links.config.json | Validate links in markdown files.                                        |
```

File: .github/workflows/README.md [hunk 3]
```diff
@@ -2,17 +2,27 @@
...
-| .github/workflows/ci-tests-pytorch.yml | Run all tests except for accelerator-specific, standalone and slow tests.                                                                                                   | CPU           |
-| .azure-pipelines/ipu-tests.yml         | Run only IPU-specific tests.                                                                                                                                                | IPU           |
-| .azure-pipelines/gpu-tests-pytorch.yml | Run all CPU and GPU-specific tests, standalone, and examples. Each standalone test needs to be run in separate processes to avoid unwanted interactions between test cases. | GPU           |
-| .azure-pipelines/gpu-benchmarks.yml    | Run speed/memory benchmarks for parity with pure PyTorch.                                                                                                                   | GPU           |
-| .github/workflows/tpu-tests.yml        | Run only TPU-specific tests. Requires that the PR title contains '\[TPU\]'                                                                                                  | TPU           |
```
Comment (carmocca) [inline .github/workflows/README.md:13]: The TPU row is missing now

File: .github/workflows/README.md [hunk 4]
```diff
@@ -22,41 +32,47 @@
...
-| .github/workflows/ci-dockers.yml     | Build docker images used for testing in CI. If run on nightly schedule, push to the [Docker Hub](https://hub.docker.com/r/pytorchlightning/pytorch_lightning). |
-| .github/workflows/ci-pkg-install.yml | Test if pytorch-lightning is successfully installed using pip.                                                                                                 |
+| workflow file                        | action                                                                                          |
+| ------------------------------------ | ----------------------------------------------------------------------------------------------- |
+| .github/workflows/docker-build.yml   | Build docker images used for testing in CI. If run on nightly schedule, push to the Docker Hub. |
+| .github/workflows/ci-pkg-install.yml | Test if pytorch-lightning is successfully installed using pip.                                  |
+| .github/workflows/ci-checkpoints.yml | Test building checkpint whci is later (when release) used for back-compatibility check.         |
```
Comment (carmocca) [inline .github/workflows/README.md:55]:
```suggestion
| .github/workflows/ci-checkpoints.yml | Build checkpoints that are will be tested on release to ensure backwards-compatibility         |
```

**Review by carmocca — APPROVED** [review]

File: .github/workflows/README.md [hunk 5]
```diff
@@ -2,17 +2,27 @@
...
+| .github/workflows/ci-tests-data.yml    | Run unit and integration tests with data pipelining.                                      | CPU         |
+| .github/workflows/ci-tests-store.yml   | Run integration tests on uploading models to cloud.                                       | CPU         |
+| .azure-pipelines/gpu-tests-fabric.yml  | Run only GPU-specific tests, standalone\*, and examples.                                  | GPU         |
+| .azure-pipelines/gpu-tests-pytorch.yml | Run only GPU-specific tests, standalone\*, and examples.                                  | GPU         |
+| .azure-pipelines/gpu-benchmarks.yml    | Run speed/memory benchmarks for parity with vanila PyTorch.                               | GPU         |
+| .github/workflows/ci-examples-app.yml  | Run integration tests with App examples.                                                  | CPU         |
+| .github/workflows/ci-flagship-apps.yml | Run end-2-end tests with full applications, including deployment to the production cloud. | CPU         |
+| .github/workflows/ci-tests-pytorch.yml | Run all tests except for accelerator-specific, standalone and slow tests.                 | CPU         |
```
Comment (Borda) [inline .github/workflows/README.md:21]:
```suggestion
| .github/workflows/ci-tests-pytorch.yml | Run all tests except for accelerator-specific, standalone and slow tests.                 | CPU         |
| .github/workflows/tpu-tests.yml | Run only TPU-specific tests. Requires that the PR title contains '\[TPU\]' | TPU |
```

**Pull request merged**